app/models/user.rb

```diff
@@ -19,6 +21,18 @@ def self.new_token
     SecureRandom.urlsafe_base64
   end
 
+  def self.create_anonimous
+    anonimous_token = SecureRandom.hex(8)
+    create username: anonimous_token, email: "#{anonimous_token}@stub",
```

Comment:
What's the difference between `create` and `create!`?
The latter throws the exception in case of validation failure, and the former just returns `false`.

So if you use `create` here, you'll ignore a potential validation error, don't create anything, and end up not finding a user where you think it should be there in the DB later on.